Add `onAfterDestroy` callback description to react docs (#16281)
ckeditor/ckeditor5: docs/installation/integrations/react.md

* `config` &ndash; The CKEditor&nbsp;5 context configuration.
* `isLayoutReady` &ndash; A property that delays the context creation when set to `false`. It creates the context and the editor children once it is `true` or unset. Useful when the CKEditor&nbsp;5 annotations or a presence list are used.
* `id` &ndash; The context ID. When this property changes, the component restarts the context with its editor and reinitializes it based on the current configuration.
* `onReady` &ndash; A function called when the context is ready and all the editors inside were initialized with the `context` instance. This callback is also called after the reinitialization of the editor if an error has occurred.

> * `onAfterDestroy` &ndash; A function invoked after the successful destruction of an editor instance that was rendered by the component. This callback is also triggered after the editor has been reinitialized after an error. It is not guaranteed that the component will still be mounted when this function is called.

Reviewer: ```suggestion
* `onAfterDestroy` &ndash; A function called after the successful destruction of an editor instance rendered by the component. This callback is also triggered after the editor has been reinitialized after an error. The component is not guaranteed to be mounted when this function is called.
```

Minor updates:

* `s/invoked/called` - otherwise, we should update other callbacks too. 
* Simplifications.

import DecoupledEditor from '@ckeditor/ckeditor5-build-decoupled-document';

class App extends Component {
	editor = null;

	render() {
		return (
			<div className="App">
				<h2>CKEditor&nbsp;5 using a custom build - decoupled editor</h2>
				<CKEditor
					onReady={ editor => {
						console.log( 'Editor is ready to use!', editor );

						// Insert the toolbar before the editable area.
						editor.ui.getEditableElement().parentElement.insertBefore(
							editor.ui.view.toolbar.element,
							editor.ui.getEditableElement()
						);

						this.editor = editor;
					} }

> 					onError={ ( error, { willEditorRestart } ) => {
> 						// If the editor is restarted, the toolbar element will be created once again.
> 						// The `onReady` callback will be called again and the new toolbar will be added.
> 						// This is why you need to remove the older toolbar.
> 						if ( willEditorRestart ) {
> 							this.editor.ui.view.toolbar.element.remove();
> 						}
> 					} }

Reviewer: ```suggestion
					onAfterDestroy={ editor => {
						editor.ui.view.toolbar.element.remove();
					} }
```

Author: I'm not sure if we should include this `onAfterDestroy` example with `editor.ui.view.toolbar.element.remove();`. It really depends on the way the users mounted `toolbar`. If it's somewhere outside of React box then it's fine but if it's in div that is managed by React then it will lead to crashes, probably. I'll leave there a comment that only external things should be removed here.

Reviewer 2: For more context: this code sample is for Decoupled Editor. This editor is characteristic in the way that it is the integrator's responsibility to insert the toolbar where they want to.
